src/common/SDKEventEmitter.ts

```diff
@@ -0,0 +1,42 @@
+import { EventEmitter } from 'events';
+
+/**
+ * A wrapper around node's events module for convenience.
+ * Adds unsubscribe method to the returned result of `addListener`
+ */
+export class SDKEventEmitter {
+  public emitter = new EventEmitter();
+
+  public addListener<
+    Type extends string,
+    Listener extends (...args: any[]) => void,
+  >(eventType: Type, listener: Listener) {
+    const subscription = this.emitter.addListener(eventType, listener);
+    const unsubscribe = () => {
+      this.emitter.removeListener(eventType, listener);
+    };
+
+    return {
+      ...subscription,
+      /**
+       * Removes this subscription from the emitter that registered it.
+       * Similar to using emitter.removeEventListener(eventName, listener)
+       * except it does not return the EventEmitter instance.
+       */
+      unsubscribe,
+      /**  @alias unsubscribe */
+      remove: unsubscribe,
```

Comment:
Why do we need to return both of these if they are the same?

Reply:
I believe I have seen this pattern in other places - some prefer one over the other. `EventEmitter` also returns different aliases like `on` etc

Reply:
I'd strongly prefer only returning one, to minimize any confusion. And, maybe _only_ returning the `unsubscribe` function (and leaving the `...subscription` out).

Reply:
@swain I would like to leave off `...subscription` - but that would cause a breaking change as the way things are today. idk if anyone would be using it right now, but probably best to deprecate over time

Any one have a preference which one do we prefer to keep? I think I like `unsubscribe` but I think most react-native api's now go with `remove`